### ISSUE TITLE
Please help with brief steps to get ASP.Net Core breeze server up

### ISSUE THREAD

**Issue by bancroftway** [Breeze/breeze.server.net#58]
It would be extremely helpful to get a list of steps to get an ASP.Net breeze server up, even if a full sample is a bit further away.

**Comment by tim-martens** [comment]
Hi @bancroftway , I created a sample project [here](https://github.com/tim-martens/Breeze.AspNet.Core) . This includes the basic setup you're looking for.

These are the steps I took to get things working:
- create a new ASP.NET Core Web Application (.NET Framework - Select the Web API project template)
- add a Class Library project (.NET Framework) to the solution for the Entity Framework
- add Entity Framework to both projects in the solution
- add your context and model classes
- add a ContextFactory class to the EF project ([check this](https://docs.microsoft.com/en-us/aspnet/core/data/entity-framework-6)

Then in the web application project:
- add a reference to the Data project
- install these nuget packages: (see [this](https://github.com/Breeze/breeze.js/issues/178#issuecomment-285463327) )
Breeze.Server.Core
Breeze.Server.Persistence
Breeze.Server.Persistence.EF6
Breeze.Server.AspNetCore.NetFramework
Breeze.Composite.AspNetCore.EF6

- modify the services configuration in Startup.cs:
`public void ConfigureServices(IServiceCollection services)
{
    // Add framework services.
    var mvcBuilder = services.AddMvc();

    mvcBuilder.AddJsonOptions(opt => {
        var ss = JsonSerializationFns.UpdateWithDefaults(opt.SerializerSettings);
        var resolver = ss.ContractResolver;
        if (resolver != null)
        {
            var res = resolver as DefaultContractResolver;
            res.NamingStrategy = null;  // <<!-- this removes the camelcasing
        }

    });

    mvcBuilder.AddMvcOptions(o => { o.Filters.Add(new GlobalExceptionFilter()); });


    services.AddScoped<SchoolContext>(_ => new SchoolContext(Configuration.GetConnectionString("SchoolContext")));
}`

- add a connectionstring the the appsettings.json
- add a breezeController:
`[Route("breeze/[controller]/[action]")]
[BreezeQueryFilter]
public class SchoolModelController : Controller
{
    private SchoolPersistenceManager PersistenceManager;

    // called via DI 
    public SchoolModelController(SchoolContext context)
    {
        PersistenceManager = new SchoolPersistenceManager(context);
    }

    [HttpGet]
    public IActionResult Metadata()
    {
        return Ok(PersistenceManager.Metadata());
    }
    [HttpPost]
    public SaveResult SaveChanges([FromBody] JObject saveBundle)
    {
        return PersistenceManager.SaveChanges(saveBundle);
    }

    [HttpGet]
    public IQueryable<Student> Students()
    {
        return PersistenceManager.Context.Students.Include("Enrollments");
    }
}

internal class SchoolPersistenceManager : EFPersistenceManager<SchoolContext>
{
    public SchoolPersistenceManager(SchoolContext context) : base(context) { }

    // additional stuff goes here. Checkout the NorthwindIBModelController of the Breeze.AspNetCore.InternalTest.sln in the breeze.server.net repo
    // https://github.com/Breeze/breeze.server.net/blob/master/Tests/Test.AspNetCore/Controllers/NorthwindIBModelController.cs
}`

Good luck!

**Comment by HansHammel** [comment]
Nope! Not working with ASP.NET Core 2.0. Libraries on nuget are build using .net 4.6.1 and therefore incomplatible...

**Comment by jasoncowan** [comment]
+1 for ASP.NET Core + EF Core 2.0. Could you pls advise if compatibility with these is on your roadmap?

**Comment by jtraband** [comment]
We are definitely planning on working on this starting in the next few weeks.  Will advise of progress and expected timeframe once we start.

**Comment by ahedreville** [comment]
Hi there is any progress on this ?😊

**Comment by jtraband** [comment]
I'm a just finishing this up now.  Should be putting it up on Nuget within a few days - but the current version ( in test) is already checked into GitHub under breeze.server.net/aspnetcore folder. ( support for both EFCore and EF6 is provided) and an example of sample servers can be found in breeze.server.net/tests/( Test.AspNetCore.EFCore and Test.AspNetCore.EF6).

**Comment by mikemichaelis** [comment]
I have all of this working in dotnet core 2.0.....ef core + webapi core + breeze aspnetcore + dotnet-cli + ef-cli

https://github.com/mikemichaelis/dnc

My persistence layer is SQL Server 2017 running in a Linux Docker container. See ya Windows........

**Comment by ahedreville** [comment]
Hi
Thanks a  lot
Now Is working on my project too..
BR

2018-02-22 20:01 GMT+01:00 Mike Michaelis <notifications@github.com>:

**Comment by gmareater** [comment]
Hello everybody
i Used  this example as an starter kit:
[https://github.com/mikemichaelis/dnc](https://github.com/mikemichaelis/dnc)
bu there is a problem when using Where clause,
I opened an SO thread (with bounty), so please anyone who can, provide a solution, thanks to all

**Comment by plwade** [comment]
Anybody got a Breeze client app working with this .net core backend? https://github.com/mikemichaelis/dnc

I would love to see the client code

**Comment by breynolds** [comment]
Is this still open because the breezejs official repo does not yet support dotnet core 2.x? Would be great to have official support. Breeze is a great help for building our applications! @mikemichaelis, we'll still use your dnc repo if we need to.

**Comment by jtraband** [comment]
Just updated both Asp.net and AspNetCore nugets. For ASP.NET Core with EF Core use the latest versions of Breeze.Core, Breeze.Persistence, Breeze.Persistence.EFCore and Breeze.AspNetCore.NetCore

**Comment by gmareater** [comment]
if you know how muuuuuchhhhhhhhhhhhhhhhhh i just waited for this message , hope it works with no bug,
i dont know about others, but i almost forget who to work without breeze, it's one  of the must have,
so please do not give up on it

**Comment by jtraband** [comment]
Thanks, please post back with any issues ... or successes.  We have run our full test suite against the new ASP.Net Core and EF Core impl, so the basics should all be there, but edge cases can obviously still occur.

**Comment by ganySA** [comment]
Any chance we could get some docs?

**Comment by steveschmitt** [comment]
I finally built a demo project, _with steps_ to create a .NET Core + EF Core server
https://github.com/Breeze/northwind-core-ng-demo
The demo is a .NET Core 2.2 web API, with Angular 8 on the front end.
I hope it helps.

**Comment by steveschmitt** [comment]
Note that the demos have been moved to https://github.com/Breeze/northwind-demo

It's a bit out of date because the latest server sample is .NET 5, but the steps to get it going are exactly the same in .NET 6 and 7.

**Comment by eqcb** [comment]
> Note that the demos have been moved to https://github.com/Breeze/northwind-demo
> 
> It's a bit out of date because the latest server sample is .NET 5, but the steps to get it going are exactly the same in .NET 6 and 7.

Hello Steve,

The steps in the .NET 5 server sample call for Startup.cs class configuration. The current templates for ASP.NET use only the program.cs. Can you guys update the sample or point to some updated example? Thanks.

**Comment by steveschmitt** [comment]
@eqcb You are right - things have changed enough that we need to update the samples.  We'll work on putting a .NET 8 sample up soon.

For an ASP.NET API, I think you should use `services.AddControllers()`.
